### ISSUE TITLE
Bing API - create form removed from Azure

### ISSUE THREAD

**Issue by codescrut** [johnwargo/learning-pwa-code#4]
Hi!

I have a problem creating the azure bing service in the chapter 3 example.

Bing search has been moved in azure with the consequence that the creation form https://portal.azure.com/#create/Microsoft.CognitiveServicesBingSearch-v7 has been removed.

Instead they refer to the following page https://docs.microsoft.com/en-us/bing/search-apis/bing-web-search/create-bing-search-service-resource.

In the KeySetting page I can see that the endpoint is https://api.bing.microsoft.com/ which does not match the endpoint expected by the server (se error dump below).

Do you know of a quick workaround for this? 

Best regards 
Lars Kappel Hansen

```text
2022-02-16T12:16:53.842Z GET /icons/android-icon-192x192.png 200 2.928 ms - 13689
Caught error
Error: Access denied due to invalid subscription key or wrong API endpoint. Make sure to provide a valid key for an active subscription and use a correct regional API endpoint for your resource.
    at D:\Locus\GitHub\pwa\chapter-03\node_modules\azure-cognitiveservices-newssearch\lib\operations\newsOperations.js:458:19
    at retryCallback (D:\Locus\GitHub\pwa\chapter-03\node_modules\ms-rest\lib\filters\systemErrorRetryPolicyFilter.js:89:9)
    at retryCallback (D:\Locus\GitHub\pwa\chapter-03\node_modules\ms-rest\lib\filters\exponentialRetryPolicyFilter.js:140:9)
    at D:\Locus\GitHub\pwa\chapter-03\node_modules\ms-rest\lib\filters\rpRegistrationFilter.js:59:14
    at handleRedirect (D:\Locus\GitHub\pwa\chapter-03\node_modules\ms-rest\lib\filters\redirectFilter.js:39:9)
    at D:\Locus\GitHub\pwa\chapter-03\node_modules\ms-rest\lib\filters\formDataFilter.js:23:14
    at Request._callback (D:\Locus\GitHub\pwa\chapter-03\node_modules\ms-rest\lib\requestPipeline.js:125:16)
    at Request.self.callback (D:\Locus\GitHub\pwa\chapter-03\node_modules\request\request.js:185:22)
    at Request.emit (node:events:520:28)
    at Request.<anonymous> (D:\Locus\GitHub\pwa\chapter-03\node_modules\request\request.js:1154:10) {
  statusCode: 401,
  request: {
    rawResponse: false,
    queryString: {},
    url: 'https://api.cognitive.microsoft.com/bing/v7.0/news/search?count=20&q=pwa',
    method: 'GET',
    headers: {
      'Content-Type': 'application/json; charset=utf-8',
      'X-BingApis-SDK': 'true',
      'Ocp-Apim-Subscription-Key': '<Key 1 supplied here>',
      'X-BingApis-SDK-Client': 'node-SDK',
      'user-agent': 'Node/v16.14.0 (x64-Windows_NT-10.0.19041) ms-rest/2.5.4 azure-cognitiveservices-newssearch/2.0.0 Azure-SDK-For-Node'
    },
    body: null
  },
  response: {
    body: '{"error":{"code":"401","message":"Access denied due to invalid subscription key or wrong API endpoint. Make sure to provide a valid key for an active subscription and use a correct regional API endpoint for your resource."}}',
    headers: {
      'content-length': '224',
      'content-type': 'application/json',
      'apim-request-id': 'd8300908-37ef-4fe1-98dd-17b73e53729f',
      date: 'Wed, 16 Feb 2022 12:16:56 GMT',
      connection: 'close'
    },
    statusCode: 401
  },
  code: '401',
  body: { _type: 'ErrorResponse' }
}
Returning catch result (3)
2022-02-16T12:16:53.873Z GET /api/news 200 173.803 ms - 51
```

**Comment by johnwargo** [comment]
@codescrut I edited your comment to make the message easier to read. 

The error message you're getting is pretty clear, its either the endpoint or your API key. Are you in the US, or somewhere else? I'm curious if that's it.

I started rebuilding the server using the new API, but never finished that work so I don't have an example with the new API running. 

A quick search gave me this, take a look: https://docs.microsoft.com/en-us/answers/questions/160048/request-to-bing-search-v7-api-unauthorized.html.

**Comment by codescrut** [comment]
@johnwargo I am situated in Denmark / Europe and the connection is made to a regional data center in Europe.

I have had an extra look at it but did not find a way to resolve the endpoint issue.
Anyway I have been involved in development of a PWA at work so in stead I use this as an example while I read on. This works out fine! 

Thanks for a good book and your time :-)

**Comment by chris-miner** [comment]
I've experienced the same problem.  The endpoint is wrong
```
https://api.cognitive.microsoft.com/bing/v7.0/news/search?count=20&q=pwa
```
should be more like:
```
https://api.bing.microsoft.com/v7.0/news/search?count=20&q=pwa
```

You can swap out the host bit by modifying the api.js file like this:
```
var client = new NewsSearchAPIClient(credentials, {
    endpoint: "https://api.bing.microsoft.com"
});
```
I didn't find a way to fix the `bing/v7.0` part, at least not without modifying the news search library.  I changed it in
```
node_modules/azure-cognitiveservices-newssearch/lib/newsSearchClient.js
```

Unlike the endpoint, it doesn't seem like it can be configured.

**Comment by johnwargo** [comment]
@chris-miner & @codescrut I fixed all the code for the new API and wrote updated docs when Microsoft first announced the change - you can find them all here: https://github.com/johnwargo/learning-pwa-code/tree/main/updated-server. Do you look at that updated code?
